nvflare/app_common/workflows/statistics_controller.py

```diff
@@ -364,19 +379,24 @@ def _wait_for_all_results(
             metric_client_received[m] = len(client_metrics[m].keys())
 
         timeout = result_wait_timeout
-        sleep_time = 5
         for m in metric_client_received:
             if requested_client_size > metric_client_received[m]:
                 t = 0
                 while t < timeout and requested_client_size > metric_client_received[m]:
-
                     if abort_signal and abort_signal.triggered:
                         return False
 
-                    self.log_info(f"not all client received the metric {m}, need to wait for {sleep_time} seconds.")
+                    msg = f"not all client received the metric {m}, need to wait for {sleep_time} seconds."
+                    if fl_ctx:
+                        self.log_info(fl_ctx, msg)
+                    else:
+                        self.logger.info(msg)
```

Comment:
suggest always required a fl_ctx
So we don't need this if else here.

Otherwise suggest remove fl_ctx from argument